### DISCUSSION TITLE
How to programmatically open/close the tooltip?

### DISCUSSION THREAD

**Discussion by duhaime** [Milkdown/milkdown#328] [Q&A] [answered]
I'm trying to figure out how to open the tooltip for processing "link" text when users click the link icon in the menu, but I'm not yet sure how to call one plugin from another. Is there an example of this behavior anywhere?

I'm thinking there may be a way to use a command, as one can do:

```
commandsCtx = ctx.get(commandsCtx)
commandsCtx.callByName('ToggleLink')
```

Though I don't think the tooltip exposes callable commands for open/close yet.

Edit: oh wow, I just learned one can use:

```
import { commands } from '@milkdown/preset-commonmark'
```

to gain access to all commands. I think one can use those commands to create and insert a link

**Comment by Saul-Mirone** [comment, marked as answer]
Since the tooltip is controlled by prosemirror selection. So what you need to do is just create a text selection programmatically. The [transaction.setSelection API in prosemirror](https://prosemirror.net/docs/ref/#state.Transaction.setSelection) can help you.

**Reply by duhaime** [reply]
Thank you!
